report/latest_map.go

```diff
@@ -90,27 +90,42 @@ func (m LatestMap) Lookup(key string) (string, bool) {
 	return v, ok
 }
 
-// LookupEntry returns the raw entry for the given key.
-func (m LatestMap) LookupEntry(key string) (string, time.Time, bool) {
+// LookupEntryExt returns the raw entry for the given key.
+func (m LatestMap) LookupEntryExt(key string) (interface{}, time.Time, bool) {
 	if m.Map == nil {
-		return "", time.Time{}, false
+		return nil, time.Time{}, false
 	}
 	value, ok := m.Map.Lookup(key)
 	if !ok {
-		return "", time.Time{}, false
+		return nil, time.Time{}, false
 	}
 	e := value.(LatestEntry)
 	return e.Value, e.Timestamp, true
 }
 
-// Set the value for the given key.
-func (m LatestMap) Set(key string, timestamp time.Time, value string) LatestMap {
+// LookupEntry returns the raw entry for the given key.
+func (m LatestMap) LookupEntry(key string) (string, time.Time, bool) {
+	str := ""
+	value, timestamp, ok := m.LookupEntryExt(key)
+	if value != nil {
+		str = fmt.Sprintf("%s", value)
+	}
+	return str, timestamp, ok
+}
+
+// SetExt sets the value for the given key.
+func (m LatestMap) SetExt(key string, timestamp time.Time, value interface{}) LatestMap {
 	if m.Map == nil {
 		m = EmptyLatestMap
 	}
 	return LatestMap{m.Map.Set(key, LatestEntry{timestamp, value})}
 }
 
+// Set the value for the given key.
+func (m LatestMap) Set(key string, timestamp time.Time, value string) LatestMap {
+	return m.SetExt(key, timestamp, value)
+}
```

Comment:
I did it that way, because I didn't want to change the LatestMap API, so I opted for leaving `Set` as is and adding a `SetExt`. I could for example rename it to `SetInterface` instead…

But I can break it if you want, no problem.